packages/core/src/fhir-deduplication/resources/condition.ts

```diff
@@ -1,115 +1,128 @@
-    const code = master.code;
-    const filtered = code?.coding?.filter(coding => {
-      const system = fetchCodingCodeOrDisplayOrSystem(coding, "system");
-      return (
-        system?.includes(SNOMED_CODE) ||
-        system?.includes(SNOMED_OID) ||
-        system?.includes(ICD_10_CODE) ||
-        system?.includes(ICD_10_OID) ||
-        system?.includes(ICD_9_CODE)
-      );
-    });
-    if (filtered && filtered.length > 0) {
-      master.code = {
-        ...code,
-        coding: filtered,
-      };
-    } else {
-      master.code = { ...code };
-      delete master.code.coding;
-    }
-    return master;
-  }
-
   for (const condition of conditions) {
     if (hasBlacklistedText(condition.code) || !isKnownCondition(condition.code)) {
       danglingReferences.add(createRef(condition));
       continue;
     }
 
     const date = getDateFromResource(condition);
-    if (!date) {
+    const { snomedCode, icd10Code } = extractCodes(condition.code);
+    const display = extractDisplayFromConcept(condition.code);
+
+    const identifiers = [
+      ...(snomedCode ? [{ snomedCode }] : []),
+      ...(icd10Code ? [{ icd10Code }] : []),
+      ...(display ? [{ display }] : []),
+    ];
+    const hasIdentifier = identifiers.length > 0;
+
+    if (!hasIdentifier) {
       danglingReferences.add(createRef(condition));
       continue;
     }
+    const getterKeys: string[] = [];
+    const setterKeys: string[] = [];
 
-    const { snomedCode, icd10Code } = extractCodes(condition.code);
-    if (icd10Code) {
-      const compKey = JSON.stringify({ icd10Code, date });
-      fillMaps(icd10Map, compKey, condition, refReplacementMap, undefined, removeOtherCodes);
-    } else if (snomedCode) {
-      const compKey = JSON.stringify({ snomedCode, date });
-      fillMaps(snomedMap, compKey, condition, refReplacementMap, undefined, removeOtherCodes);
+    if (date) {
+      // flagging the condition with each unique identifier + date
+      setterKeys.push(...createKeysFromObjectArray({ date }, identifiers));
+      // flagging the condition with each unique identifier + 1 date bit
+      setterKeys.push(...createKeysFromObjectArrayAndBits(identifiers, [1]));
+
+      // the condition will dedup using each unique identifier with the same date,
+      getterKeys.push(...createKeysFromObjectArray({ date }, identifiers));
+      // the condition will dedup using each unique identifier + 0 date bit
+      getterKeys.push(...createKeysFromObjectArrayAndBits(identifiers, [0]));
+    }
+
+    if (!date) {
+      // flagging the condition with each unique identifier + 0 date bit
+      setterKeys.push(...createKeysFromObjectArrayAndBits(identifiers, [0]));
+
+      // the condition will dedup using each unique identifier with 0 or 1 date bit
+      getterKeys.push(...createKeysFromObjectArrayAndBits(identifiers, [0]));
+      getterKeys.push(...createKeysFromObjectArrayAndBits(identifiers, [1]));
```

Comment:
nit: maybe store those so we don'e need to rebuild on the diff places we use it?

Reply:
`firstKeys` and `secondKeys`? 😃 

NABD if we don't have this running in large loops, but if we do, especially if they're nested and we have a lot of identifiers, these things can save some considerable CPU.